awscli/customizations/history/list.py

```diff
@@ -0,0 +1,491 @@
+# Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import json
+import datetime
+
+import colorama
+
+from botocore.compat import OrderedDict
+
+from awscli.customizations.history.commands import HistorySubcommand
+
+
+class ListCommand(HistorySubcommand):
+    NAME = 'list'
+    DESCRIPTION = (
+        'Shows a list of previously run commands and their command_ids. '
+        'Each row shows only a bare minimum of details including the '
+        'command_id, date, arguments and return code. You can use the '
+        '``history show`` with the command_id to see more details about '
+        'a particular entry.'
+    )
+    _COL_WIDTHS = OrderedDict([
+        ('Command Id', 38),
+        ('Time', 24),
+        ('Arguments', 50),
+        ('Return Code', 13)
+    ])
+
+    def _run_main(self, parsed_args, parsed_globals):
+        self._connect_to_history_db()
+        raw_records = self._db_reader.iter_all_records()
+        records = RecordAdapter(raw_records)
+        if not records.has_next():
+            raise RuntimeError('No recorded history.')
+
+        with self._get_output_stream('less -SR') as output_stream:
```

Comment:
I just realized this from reading the tests but I do not think we can hard code it like this to always use `less -SR` as it will not work for windows. Mainly because windows does not have `less` installed, but I would recommend checking before changing it. It is not an issue though if that is case. I say in the `compat` module we just add the following logic that we can import into the `list` command:
```python
is_windows:
     default_pager = 'more'
else:
     default_pager = 'less -R'
```
From there we can just check if the default pager we got back is less and just clobber it with our own in `less`